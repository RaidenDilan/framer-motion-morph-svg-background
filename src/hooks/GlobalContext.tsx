import React, { createContext, ReactNode, useContext, useState } from 'react';

interface GlobalContextInterface {
  isLoading: boolean;
}

export const defaultValues: GlobalContextInterface = {
  isLoading: false,
};

export const GlobalCtx = createContext<GlobalContextInterface>(defaultValues);

export const GlobalProvider = ({ children }: { children: ReactNode }) => {
  const [isLoading, setIsLoading] = useState<boolean>(false);

  return (
    <GlobalCtx.Provider value={{ isLoading }}>{children}</GlobalCtx.Provider>
  );
};

export const useGlobalContext = () => useContext(GlobalCtx);
