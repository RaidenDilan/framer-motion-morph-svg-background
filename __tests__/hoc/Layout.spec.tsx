import React from 'react';
import { create } from 'react-test-renderer';
import { render } from '@testing-library/react';

import { GlobalCtx, defaultValues } from '../../src/hooks/GlobalContext';

const LayoutRoute = () => <React.Fragment />;
const ProtectedRoute = () => <React.Fragment />;

const navigate = jest.fn();

jest.mock('react-router-dom', () => {
  const requireActual = jest.requireActual('react-router-dom');
  return {
    ...requireActual,
    useNavigate: () => navigate,
  };
});

describe('Layout', () => {
  it('renders layout onboarding without token', () => {
    const { asFragment } = render(
      <GlobalCtx.Provider value={{ ...defaultValues }}>
        <LayoutRoute />
      </GlobalCtx.Provider>,
    );

    expect(asFragment()).toMatchSnapshot();

    expect(navigate.mock.calls).toMatchSnapshot();
  });

  it('renders layout with isLoading', () => {
    expect(
      create(
        <GlobalCtx.Provider value={{ ...defaultValues, isLoading: true }}>
          <LayoutRoute />
        </GlobalCtx.Provider>,
      ),
    ).toMatchSnapshot();

    expect(navigate.mock.calls).toMatchSnapshot();
  });

  it('renders protected route without token', () => {
    expect(
      create(
        <GlobalCtx.Provider value={{ ...defaultValues }}>
          <ProtectedRoute />
        </GlobalCtx.Provider>,
      ),
    ).toMatchSnapshot();

    expect(navigate.mock.calls).toMatchSnapshot();
  });

  it('renders protected route with isLoading', () => {
    const { asFragment } = render(
      <GlobalCtx.Provider value={{ ...defaultValues, isLoading: true }}>
        <LayoutRoute />
      </GlobalCtx.Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
    expect(navigate.mock.calls).toMatchSnapshot();
  });
});
