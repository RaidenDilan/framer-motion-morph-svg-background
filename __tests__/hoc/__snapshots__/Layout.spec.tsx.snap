// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders layout onboarding without token 1`] = `<DocumentFragment />`;

exports[`Layout renders layout onboarding without token 2`] = `Array []`;

exports[`Layout renders layout with isLoading 1`] = `null`;

exports[`Layout renders layout with isLoading 2`] = `Array []`;

exports[`Layout renders protected route with isLoading 1`] = `<DocumentFragment />`;

exports[`Layout renders protected route with isLoading 2`] = `Array []`;

exports[`Layout renders protected route without token 1`] = `null`;

exports[`Layout renders protected route without token 2`] = `Array []`;
